src/NodaTime/LocalDate.cs

```diff
@@ -581,6 +581,24 @@ int IComparable.CompareTo(object obj)
         public bool Equals(LocalDate other) => this == other;
 
         /// <summary>
+        /// Resolves this local date into a <see cref="ZonedDateTime"/> in the given time zone representing the
+        /// start of this date in the given zone.
+        /// </summary>
+        /// <remarks>
+        /// This is a convenience method for calling <see cref="DateTimeZone.AtStartOfDay(LocalDate)"/>.
+        /// </remarks>
+        /// <param name="zone">The time zone to map this local date into</param>
+        /// <exception cref="SkippedTimeException">The entire day was skipped due to a very large time zone transition.
+        /// (This is extremely rare.)</exception>
+        /// <returns>The <see cref="ZonedDateTime"/> representing the earliest time on this date, in the given time zone.</returns>
+        [Pure]
+        public ZonedDateTime InZone([NotNull] DateTimeZone zone)
```

Comment:
Since we already have `zone.AtStartOfDay(localDate)`, I'm actually thinking that `localDate.AtStartOfDay(zone)` is fairly sensible.

(No, it doesn't have 'zone' in the name.)

The only problem is that `LocalDate.AtMidnight()` exists, returning a `LocalDateTime`, and it's not obvious how this differs.